neutron/clientheaders.go

```diff
@@ -0,0 +1,63 @@
+package neutron
+
+import (
+	"net/http"
+
+	goosehttp "gopkg.in/goose.v2/http"
+)
+
+// NeutronHeaders creates a set of http.Headers from the given arguments passed
+// in.
+// In this case it applies the headers passed in first, then sets the following:
+//  - X-Auth-Token
+//  - Content-Type
+//  - Accept
+//  - User-Agent
+//
+func NeutronHeaders(method string, extraHeaders http.Header, contentType, authToken string, payloadExists bool) http.Header {
+	headers := goosehttp.BasicHeaders()
+
+	if authToken != "" {
+		headers.Set("X-Auth-Token", authToken)
+	}
+
+	// Officially we should also take into account the method, as we should not
+	// be applying this to every request.
+	if payloadExists {
+		headers.Add("Content-Type", contentType)
+	}
+
+	// According to the REST specs, the following should be considered the
+	// correct implementation of requests. Openstack implementation follow
+	// these specs and will return errors if certain headers are pressent.
+	//
```

Comment:
We don't explicitly follow this, because of the `Content-Type`, which used the `payloadExists` rather than the method. Instead it should really be (pseudo code):

```go
switch method {
case "POST", "PUT", "PATCH":
	if payloadExists {
		headers.Add("Content-Type", contentType)
	}
default:
	// We don't care here.
}
```